x-pack/plugin/mapper-flattened/src/main/java/org/elasticsearch/xpack/flattened/mapper/FlatObjectFieldMapper.java

```diff
@@ -471,15 +471,7 @@ public Object valueForDisplay(Object value) {
 
         @Override
         public ValueFetcher valueFetcher(MapperService mapperService, SearchLookup searchLookup, String format) {
-            if (format != null) {
-                throw new IllegalArgumentException("Field [" + name() + "] of type [" + typeName() + "] doesn't support formats.");
-            }
-            return new SourceValueFetcher(name(), mapperService, nullValue) {
```

Comment:
Through this refactor I discovered that we were incorrectly applying `nullValue` to the root flattened field. I'll open another PR to clean up value fetching for flattened fields, which should be reviewed/ merged first.